### DISCUSSION TITLE
ADetailer prompt based on the main prompt but changed

### DISCUSSION THREAD

**Discussion by SirMathelot** [Bing-su/adetailer#287] [Ideas]
It'd be cool if there was a way to have the ADetailer prompt be generated based on the main prompt but for example omit some parts of the main prompt. Here's an example: (I'm using dynamic prompts for randomization)

if my main prompt is:
{Asian|European} woman wearing high heels

I want the ADetailer prompt be:
"European woman"
or
"Asian woman"

based on which one is picked in the main prompt by dynamic prompts. I don't want to include the high heels part in the ADetailer prompt because it's useless there because it's only inpainting the face. Or for example if I don't want to use some LORA in the ADetailer but want the rest of the main prompt. You could for example surround the "wearing high heels" with some brackets to indicate that this part shouldn't be used in the ADetailer prompt.

**Comment by ghost** [comment]
I don't have a solution to this but I'm in the same boat and it's really frustrating!

**Comment by Symbiomatrix** [comment]
Might be doable. To my understanding, most of adetailer runs in the postprocess_image callback, with a bit of parm collection in the [process call](https://github.com/Bing-su/adetailer/blob/22602536535f850dae19fb4886e0dc8febf88f09/scripts/!adetailer.py#L523). The parm collection should therefore with relative ease employ the same or simpler parsing we do in regional prompter's process, with some keyword, to clone or extract part of the main prompt for one of the adetailer calls.
To address the matter of dynamic prompts, that extension would then have to run before adetailer so it can replace this randomisation clause before adetailer takes it aside. Had a similar issue with wildcards a while back. Last I checked, the only way to force order of extension calls was by their alphabetical name, and even of that I'm not entirely sure (though vlad was working on a priority setting), haven't really checked whether anything's changed as yet but I'm dubious as by the looks of scriptrunner's code that appears to be the case still.
